### PR TITLE
IGAPP-771: Geo json properties custom type

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: api-client/src/maps/index.ts [hunk 1]
```diff
@@ -11,6 +11,18 @@ export type MapViewViewport = {
   zoom: number
 }
 
+/**
+ * Override existing GeoJsonProperties from types/geojson to be more precise
+ */
+export type GeoJsonPoiProperties = {
```
Comment (klinzo) [inline api-client/src/maps/index.ts:17]: Maybe we could create aliases for `Feature<Point, GeoJsonPoiProperties>` and `FeatureCollection<Point, GeoJsonPoiProperties` as well? We are using them really often and I think simpler types would be nice

**Review by klinzo — APPROVED** [review]

File: api-client/src/endpoints/createPOIsEndpoint.ts [hunk 2]
```diff
@@ -20,29 +20,32 @@ export default (baseUrl: string): Endpoint<ParamsType, Array<PoiModel>> =>
...
-              country: poi.location.country,
-              latitude: poi.location.latitude,
-              longitude: poi.location.longitude
-            }),
-            lastUpdate: moment.tz(poi.modified_gmt, 'GMT'),
-            hash: poi.hash
-          })
-        })
+        json
+          .filter(poi => poi.location.id && poi.location.name && poi.location.latitude && poi.location.longitude)
```
Comment (Taggotty) [inline api-client/src/endpoints/createPOIsEndpoint.ts:24]: I think we still want to show locations without longitude and latitude in the Pois list

**Review by Taggotty — COMMENTED** [review]
In POIListItem you'll need to change `poi.location` to `poi.location.location`
And `feature.properties` doesn't seem to be optional anymore. Therefore, replace all occurences of `feature.properties?` with `feature.properties`

File: api-client/src/endpoints/createPOIsEndpoint.ts (hunk 2 above)
Comment (klinzo) [reply api-client/src/endpoints/createPOIsEndpoint.ts:24]: Hmm not sure about that. I think every POI that is relevant should have a longitude/latitude.

Comment (f1sh1918) [reply api-client/src/endpoints/createPOIsEndpoint.ts:24]: i'm with @klinzo the poi feature does not make any sense for me without displaying them on the map and have an address. i think with the cms plugin a poi that has an address also has long/lat

File: api-client/src/maps/index.ts (hunk 1 above)
Comment (f1sh1918) [reply api-client/src/maps/index.ts:17]: is that what you have expected?

```js
// aliases for Features and FeatureCollections using custom GeoJsonProperties and Point
export type PoiFeature = Feature<Point, GeoJsonPoiProperties>
export type PoiFeatureCollection = FeatureCollection<Point, GeoJsonPoiProperties>
```

Comment (klinzo) [reply api-client/src/maps/index.ts:17]: Yep, would be nice!

**Comment by f1sh1918** [comment]
> In POIListItem you'll need to change `poi.location` to `poi.location.location` And `feature.properties` doesn't seem to be optional anymore. Therefore, replace all occurences of `feature.properties?` with `feature.properties`

Updated pr so pls recheck

File: web/src/components/MapView.tsx [hunk 3]
```diff
@@ -38,19 +37,19 @@ const geolocateControlStyle: React.CSSProperties = {
...
-  featureCollection: FeatureCollection<Point>
+  featureCollection: PoiFeatureCollection
 }
 
 const MapView: React.FunctionComponent<MapViewProps> = (props: MapViewProps): ReactElement => {
   const { featureCollection, bboxViewport } = props
   const [viewport, setViewport] = useState<MapViewViewport>(bboxViewport)
   const [showPopup, togglePopup] = React.useState<boolean>(false)
-  const [currentFeature, setCurrentFeature] = React.useState<Feature<Point> | null>(null)
+  const [currentFeature, setCurrentFeature] = React.useState<PoiFeature | null>(null)
```
Comment (Taggotty) [inline web/src/components/MapView.tsx:47]: Try staying consistent with `React.useState` and `useState`

**Review by Taggotty — APPROVED** [review]

File: web/src/components/MapPopup.tsx [hunk 4]
```diff
@@ -1,15 +1,17 @@
-import { GeoJsonProperties, Position } from 'geojson'
+import { Position } from 'geojson'
 import React, { ReactElement } from 'react'
 import { Popup } from 'react-map-gl'
 import styled from 'styled-components'
 
+import { GeoJsonPoiProperties } from 'api-client/src'
```
Comment (klinzo) [inline web/src/components/MapPopup.tsx:6]:
```suggestion
import { GeoJsonPoiProperties } from 'api-client'
```

**Pull request merged**